src/main/java/team/chisel/ctm/client/model/ModelCTM.java

```diff
@@ -66,7 +67,9 @@ public class ModelCTM implements IModelCTM {
     protected Map<Pair<Integer, ResourceLocation>, ICTMTexture<?>> textureOverrides;
 
     private final Collection<ResourceLocation> textureDependencies;
-    
+
+    private final Set<RenderType> extraLayers = new HashSet<>();
+    private final Set<RenderType> extraLayersView = Collections.unmodifiableSet(extraLayers);
```

Comment:
Is there a purpose to keeping the old byte field? It was an optimization over exactly this sort of data structure, but since canRenderInLayer is no longer used...

Reply:
Mainly because I was unsure about how the implementation to `canRenderInLayer` would change and figured it might as well stay until that may be removed, but thinking about it I can remove the byte field and change the unused `canRenderInLayer` to be implemented as such:
```java
if (getExtraLayers(state).contains(layer)) {
    return true;
}
BakedModel bakedModel = Minecraft.getInstance().getModelManager().getBlockModelShaper().getBlockModel(state);
return bakedModel.getRenderTypes(state, RandomSource.create(42), ModelData.EMPTY).contains(layer);
```

and that way it would at least be returning mostly accurate results (in theory replacing the byte with a boolean so that it can keep track of if there are some layers with no custom render lay would make it even more accurate but probably isn't needed. Would you like me to make the proposed change to the implementation of `canRenderInLayer` so that I can remove the byte field and also have it so that it returns the proper result for cases it currently does not?